main.go

```diff
@@ -44,24 +43,33 @@ func main() {
 	}
 
 	// Get new finder instance
-	instance, err := finder.NewFinder(args, pci2ResMap)
-	if err != nil {
-		log.Fatalf("Failed to initialize Finder instance: %v", err)
+	var finderInstance finder.Finder
+	if args.Source == "cri"{
+		finderInstance, err = finder.NewCRIFinder(args, pci2ResMap)
+		if err != nil {
+			log.Fatalf("Failed to initialize Finder instance: %v", err)
+		}
+	}else if args.Source == "pod-resource-api"{
+		finderInstance, err = finder.NewPodResourceFinder(args, pci2ResMap)
+		if err != nil {
+			log.Fatalf("Failed to initialize Finder instance: %v", err)
+		}
```

Comment:
no need to duplicate this check: just move it outside the if/else

Reply:
Done